ngraph/core/src/op/asinh.cpp

```diff
@@ -32,25 +53,22 @@ shared_ptr<Node> op::v3::Asinh::clone_with_new_inputs(const OutputVector& new_ar
 namespace asinhop
 {
     template <element::Type_t ET>
-    bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out)
+    inline bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out, const size_t count)
     {
-        runtime::reference::asinh(
-            arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), shape_size(arg0->get_shape()));
+        runtime::reference::asinh(arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), count);
         return true;
     }
 
     bool evaluate_asinh(const HostTensorPtr& arg0, const HostTensorPtr& out)
     {
         bool rc = true;
+        size_t count = shape_size(arg0->get_shape());
         out->set_unary(arg0);
+
         switch (arg0->get_element_type())
         {
-            NGRAPH_TYPE_CASE(evaluate_asinh, i32, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_asinh, i64, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_asinh, u32, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_asinh, u64, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_asinh, f16, arg0, out);
-            NGRAPH_TYPE_CASE(evaluate_asinh, f32, arg0, out);
+            NGRAPH_TYPE_CASE(evaluate_asinh, f16, arg0, out, count);
```

Comment:
Why did you reduce the number of supported types?

Reply:
The spec (even before re-format) says: "T: any floating point type"

Reply:
I think we should not remove type support in evaluate method. In this case instead of restricting the types according to spec, I believe we should change spec to have any supported types in Types section

Reply:
So we should

1. change spec from "T: any floating-point type" to "T: any numeric type"
2. keep the types here as i32/i64/u32/u64/f16/f32 even these types are only subset of the "numeric type" (i8/i4... are not included)

Am I correct?